Makefile

```diff
@@ -1,83 +1,190 @@
-include ./config.mk
+include ./config/.env
 
-TAGNAME ?= default
+TAGNAME := $(if $(TAGNAME),$(TAGNAME),distros)
 
 .PHONY: test-env-up
 test-env-up:
-	docker build . -q -f ./scripts/Dockerfile.build -t acceptance-test-${TAGNAME}
+	@docker build . -q -f ./scripts/Dockerfile.build -t acceptance-test-${TAGNAME}
 
-.PHONY: test-run
 test-run:
-	@docker run -d --name acceptance-test-${IMGNAME} -t \
-      -e AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}" \
-      -e AWS_SECRET_ACCESS_KEY="${AWS_SECRET_ACCESS_KEY}" \
-      -v ${ACCESS_KEY_LOCAL}:/go/src/github.com/rancher/distros-test-framework/config/.ssh/aws_key.pem \
-      acceptance-test-${TAGNAME} sh -c 'cd ./entrypoint; \
-    if [ -n "${TESTDIR}" ]; then \
-        if [ "${TESTDIR}" = "upgradecluster" ]; then \
-            go test -timeout=45m -v ./upgradecluster/... -installVersionOrCommit "${INSTALLTYPE}"; \
-        elif [ "${TESTDIR}" = "versionbump" ]; then \
-            go test -timeout=45m -v -tags=versionbump ./versionbump/... -cmd "${CMD}" -expectedValue "${VALUE}" \
-            -expectedValueUpgrade "${VALUEUPGRADED}" -installVersionOrCommit "${INSTALLTYPE}" -channel "${CHANNEL}" -testCase "${TESTCASE}" \
-            -deployWorkload "${DEPLOYWORKLOAD}" -workloadName "${WORKLOADNAME}" -description "${DESCRIPTION}"; \
-		elif [ "${TESTDIR}" = "mixedoscluster" ]; then \
-            go test -timeout=45m -v -tags=mixedos ./mixedoscluster/...; \
-        fi; \
-    elif [ -z "${TESTDIR}" ]; then \
-        go test -timeout=45m -v ./createcluster/...; \
-    fi;'
+	@if [ -z "$${IMGNAME}" ]; then IMGNAME=${IMGNAME}; fi; \
+	if [ -z "$${TAGNAME}" ]; then TAGNAME=${TAGNAME}; fi; \
+	  docker run -dt --name acceptance-test-$${IMGNAME} \
+	  -e AWS_ACCESS_KEY_ID=$${AWS_ACCESS_KEY_ID} \
+	  -e AWS_SECRET_ACCESS_KEY=$${AWS_SECRET_ACCESS_KEY} \
+	  --env-file ./config/.env \
+	  -v ${ACCESS_KEY_LOCAL}:/go/src/github.com/rancher/distros-test-framework/config/.ssh/aws_key.pem \
+	  -v ./scripts/test-runner.sh:/go/src/github.com/rancher/distros-test-framework/scripts/test-runner.sh \
+	  acceptance-test-${TAGNAME} && \
+	  make test-logs USE=IMGNAME acceptance-test-${IMGNAME}
+
+test-run-state:
+	DOCKERCOMMIT=$$? \
+	CONTAINER_ID=$(shell docker ps -a -q --filter ancestor=acceptance-test-${TAGNAME} | head -n 1); \
+    	if [ -z "$$CONTAINER_ID" ]; then \
+    		echo "No matching container found."; \
+    		exit 1; \
+    	else \
+    		docker commit $$CONTAINER_ID teststate:latest; \
+    		if [ $$DOCKERCOMMIT -eq 0 ]; then \
+    			docker run -dt --name acceptance-test-${TESTSTATE} --env-file ./config/.env \
+    			-e AWS_ACCESS_KEY_ID=$${AWS_ACCESS_KEY_ID} \
+    			-e AWS_SECRET_ACCESS_KEY=$${AWS_SECRET_ACCESS_KEY} \
```

Comment:
Can we standardize the way the variables are accessed? Example $$var ${Var} $Var Based on the right thing to use

Reply:
sure